config-page/settingsKeys.txt
SETTING_TIME_COLOR,
SETTING_SUBTEXT_PRIMARY_COLOR,
SETTING_SUBTEXT_SECONDARY_COLOR,
SETTING_BG_COLOR,
SETTING_PIP_COLOR_PRIMARY,
SETTING_PIP_COLOR_SECONDARY,
SETTING_RING_STROKE_COLOR,
SETTING_RING_NIGHT_COLOR,
SETTING_RING_DAY_COLOR,
SETTING_RING_SUNRISE_COLOR,
SETTING_RING_SUNSET_COLOR,
SETTING_SUN_STROKE_COLOR,
SETTING_SUN_FILL_COLOR,
SETTING_NIGHT_TIME_COLOR,
SETTING_NIGHT_SUBTEXT_PRIMARY_COLOR,
SETTING_NIGHT_SUBTEXT_SECONDARY_COLOR,
SETTING_NIGHT_BG_COLOR,
SETTING_NIGHT_PIP_COLOR_PRIMARY,
SETTING_NIGHT_PIP_COLOR_SECONDARY,
SETTING_NIGHT_RING_STROKE_COLOR,
SETTING_NIGHT_RING_NIGHT_COLOR,
SETTING_NIGHT_RING_DAY_COLOR,
SETTING_NIGHT_RING_SUNRISE_COLOR,
SETTING_NIGHT_RING_SUNSET_COLOR,
SETTING_NIGHT_SUN_STROKE_COLOR,
SETTING_NIGHT_SUN_FILL_COLOR,
SETTING_USE_LARGE_FONTS,
SETTING_USE_NIGHT_THEME,
SETTING_PIP_VISIBILITY,
SETTING_SHOW_LEADING_ZERO

Releases/Changelog.md
# TimeStyle Versions

## Release-7.5
_Published on 2025-08-02 01:40:00 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk, Diorite, Emery

Release Notes:
v7.5 ("Risen From The Grave")
- New feature: UV Index widget, allowing you to determine precisely how much of a problem it is that you forgot to put on sunscreen
- Improvement: Swatch™ Internet Time widget is now available for classic Pebbles, allowing OG users to finally know what time it is on the internet
- Improvement: Alternate Time Zone widget now supports selecting a time zone instead of using a confusing hourly offset (thanks FlynnD273!)
- Improvement: The sleep widget has been restored and is now selectable as independent widget
- Fix: The manual location input for weather in the settings page works again, and no longer depends on Google to work
- Change: Replaced OpenWeatherMap weather data provider with OpenMeteo, which should result in higher accuracy and also the ability to get cool new data such as the previously-mentioned UV Index
- Change: Removed WeatherUnderground as a provider option, since it is no longer possible to create API keys for it
- Change: Default theme on black & white watches is now higher contrast
- Change: Weather widgets are used by default because that's what everyone used anyway

## Release-7.1
_Published on 2017-01-11 15:48:12 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk, Diorite, Emery

Release Notes:
v7.1
* Oops left in a debug line that broke the config page. Fixed now.

## Release-7.0
_Published on 2017-01-11 15:06:17 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk, Diorite, Emery

Release Notes:
Version 7.0 ("The Apocalypse Update")
	• Support for Time 2, so that when you pay $2000 for a prototype Time 2 on Ebay you'll be able to run TimeStyle on it
	• Added option to hide the Bluetooth disconnection icon, for when the Pebble app is but a distant memory
		○ (Thanks to plarus on GitHub for this contribution!)
	• Fixed visual glitch when LECO is selected and leading zero is hidden
	• Added the Hebrew language (language pack required) so that you'll be able to converse with the four horsemen of the apocalypse

## Release-6.11
_Published on 2016-11-25 13:33:42 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk, Diorite

Release Notes:
v6.11
 - Vibration settings now respect watch quiet time setting

## Release-6.10
_Published on 2016-11-03 14:07:28 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk, Diorite

Release Notes:
v6.10
 • Removed sleep function temporarily, until Pebble gets their act together and fixes the sleep bug
 • Add Korean language
 • Fixes that allow building on latest SDK

## Release-6.9
_Published on 2016-10-24 02:32:09 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk, Diorite

Release Notes:
v6.9:
 	• New language option: 日本語 (Japanese, requires language pack)
	• New sidebar widget: heart rate monitor (for Pebble 2 only)
	• Improved LECO font metrics to better match "classic" appearance
	• Fixed a bug where antialiasing would fail to re-enable when changing fonts
	• Fixed a bug where certain combinations of widgets would overlap in Quick View mode

## Release-6.8
_Published on 2016-09-30 20:46:05 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk, Diorite

Release Notes:
v6.8
 - Fixed an issue where the center widget wouldn't display if large fonts were enabled

## Release-6.7
_Published on 2016-09-30 02:19:43 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk, Diorite

Release Notes:
v6.7
* Slight fix for off-center time on round watches

## Release-6.6
_Published on 2016-09-30 01:25:28 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk, Diorite

Release Notes:
v6.6
- At last! Timeline Peek support!
- Improved font metrics for round watches
- Sidebar now defaults to grey on black and white watches
- Menu icon no longer has white corners on black and white watches

## Release-6.5
_Published on 2016-09-11 21:31:02 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk, Diorite

Release Notes:
v6.5 "The calm before the storm"
- Preliminary support for the Pebble 2
- Added Gallacian language

## Release-6.4
_Published on 2016-06-22 02:42:14 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk

Release Notes:
v6.4
 * Fixed issue from 6.3 update that broke GPS weather

## Release-6.3
_Published on 2016-06-20 01:19:52 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk

Release Notes:
v6.3
 * Fixed issue where Weather Underground wouldn't work for certain locations

## Release-6.2
_Published on 2016-05-30 17:30:37 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk

Release Notes:
v6.2
 - Fixed an issue where only "day" weather icons were shown if Weather Underground was used

## Release-6.1
_Published on 2016-05-29 14:50:51 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk

Release Notes:
v6.1
 - Fix for potential issue where weather updates could become erroneously disabled

## Release-6.0
_Published on 2016-05-29 01:09:18 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk

Release Notes:
	• New! Alternative weather data source
		○ Want more accurate weather data? Now you can choose Weather Underground, as long as you provide an API key!
	• Fixed a bug where OWM forecasts often had the same high and low temperature
	• Hourly vibe is now a double pulse, to distinguish it from notifications
	• Battery now shows "5%" instead of "0%" when at lowest level, to match official battery meter appearance
	• Percent sign now shows to the left of the number when language is set to Turkish
Month of "May" now fits slightly better in the sidebar

## Release-5.13
_Published on 2016-04-02 18:44:35 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk

Release Notes:
v5.13
 - Fix for an issue where the weather failed to update (thanks, IguanaBen!)

## Release-5.12
_Published on 2016-03-28 14:27:00 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk

Release Notes:
v5.12
 - Restored weather forecast functionality via dark magickry

## Release-5.11
_Published on 2016-03-25 04:01:36 UTC_

Hardware Platforms:
Aplite, Basalt, Chalk

Release Notes:

src/settings.h
#pragma once
#include <pebble.h>
#include "sidebar_widgets.h"

#define LEGACY_SETTINGS_PERSIST_KEY 3
#define SETTINGS_VERSION_KEY 4

// settings "version" for app version 4.0
#define CURRENT_SETTINGS_VERSION 4

typedef struct {
  // color settings
  GColor timeColor;
  GColor timeBgColor;
  GColor sidebarColor;
  GColor sidebarTextColor;

  // general settings
  uint8_t languageId;
  bool showLeadingZero;
  bool clockFontId;

  // vibration settings
  bool btVibe;
  bool hourlyVibe;

  // sidebar settings
  SidebarWidgetType widgets[3];
  bool sidebarOnLeft;
  bool useLargeFonts;

  // weather widget settings
  bool disableWeather;
  bool useMetric;

  // battery meter widget settings
  bool showBatteryPct;
  bool onlyShowBatteryWhenLow;
} Settings;

extern Settings globalSettings;

// persistent storage keys for each setting

// color settings
#define SETTING_TIME_COLOR_KEY            17
#define SETTING_TIME_BG_COLOR_KEY         18
#define SETTING_SIDEBAR_COLOR_KEY         19
#define SETTING_SIDEBAR_TEXT_COLOR_KEY    20

// general settings
#define SETTING_LANGUAGE_ID_KEY           25
#define SETTING_LEADING_ZERO_KEY          10
#define SETTING_CLOCK_FONT_ID_KEY         13

// vibration settings
#define SETTING_BT_VIBE_KEY               23
#define SETTING_HOURLY_VIBE_KEY           14

// sidebar settings
#define SETTING_SIDEBAR_WIDGET0_KEY       26
#define SETTING_SIDEBAR_WIDGET1_KEY       27
#define SETTING_SIDEBAR_WIDGET2_KEY       28
#define SETTING_SIDEBAR_LEFT_KEY          22
#define SETTING_USE_LARGE_FONTS_KEY       16

// weather widget settings
#define SETTING_DISABLE_WEATHER_KEY       12
#define SETTING_USE_METRIC_KEY            21

// battery meter widget settings
#define SETTING_BATTERY_ONLY_WHEN_LOW_KEY 15
#define SETTING_SHOW_BATTERY_PCT_KEY      11

// deprecated settings (to be deleted in next version)
#define SETTING_SHOW_BATTERY_METER_KEY    24

void Settings_init();
void Settings_deinit();
void Settings_loadFromStorage();
void Settings_saveToStorage();
